src/conn/conn_compact.c

```diff
@@ -21,21 +22,256 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_list_insert --
+ *     Insert compaction statistics for a file into the background compact list.
+ */
+static void
+__background_compact_list_insert(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    ++(conn)->background_compact.file_count;
+}
+
+/*
+ * __background_compact_list_remove --
+ *     Remove compaction statistics for a file from the background compact list.
+ */
+static void
+__background_compact_list_remove(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_REMOVE(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_REMOVE(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    WT_ASSERT(session, (conn)->background_compact.file_count > 0);
+    --(conn)->background_compact.file_count;
+
+    __wt_free(session, compact_stat->uri);
+    __wt_free(session, compact_stat);
+}
+
+/*
+ * __background_compact_get_stat --
+ *     Get the statistics for the given uri.
```

Comment:
Could you update this comment to describe why the pair of URI and id are necessary? I know there is an embedded comment, but I think it would be useful here as well.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9541/commits/112ccd9c97af7114df00e6b89c772d59b5d29444.